src/governance-classes.cpp

```diff
@@ -0,0 +1,492 @@
+{
+    DBG( cout << "CGovernanceTriggerManager::AddNewTrigger: Start" << endl; );
+    AssertLockHeld(governance.cs);
+
+    // IF WE ALREADY HAVE THIS HASH, RETURN
+    if(mapTrigger.count(nHash))  {
+        DBG( 
+            cout << "CGovernanceTriggerManager::AddNewTrigger: Already have hash"
+                 << ", nHash = " << nHash.GetHex()
+                 << ", count = " << mapTrigger.count(nHash)
+                 << ", mapTrigger.size() = " << mapTrigger.size()
+                 << endl; );
+        return false;
+    }
+
+    CSuperblock_sptr superblock(new CSuperblock(nHash));
+
+    if(superblock->GetErrorState()) {
+        DBG( cout << "AddNewTrigger: error in superblock: " << superblock->GetErrorMessage() << endl; );
+        LogPrint("superblock", "CGovernanceTriggerManager::AddNewTrigger: Error creating superblock: %s\n", superblock->GetErrorMessage());
+        return false;
+    }
+
+    superblock->SetStatus(SEEN_OBJECT_IS_VALID);
+
+    DBG( cout << "CGovernanceTriggerManager::AddNewTrigger: Inserting trigger" << endl; );
+    mapTrigger.insert(make_pair(nHash, superblock));
+
+    DBG( cout << "CGovernanceTriggerManager::AddNewTrigger: End" << endl; );
+
+    return true;
+}
+
+/**
+*
+*   Clean And Remove
+*
+*/
+
+void CGovernanceTriggerManager::CleanAndRemove()
+{
+    DBG( cout << "CGovernanceTriggerManager::CleanAndRemove: Start" << endl; );
+    AssertLockHeld(governance.cs);
+
+    // LOOK AT THESE OBJECTS AND COMPILE A VALID LIST OF TRIGGERS
+    for(trigger_m_it it = mapTrigger.begin(); it != mapTrigger.end(); ++it)  {
+        //int nNewStatus = -1;
+        CGovernanceObject* pObj = governance.FindGovernanceObject((*it).first);
+        if(!pObj)  {
+            continue;
+        }
+        CSuperblock_sptr& superblock = it->second;
+        if(!superblock)  {
+            continue;
+        }
+        // IF THIS ISN'T A TRIGGER, WHY ARE WE HERE?
+        if(pObj->GetObjectType() != GOVERNANCE_OBJECT_TRIGGER) {
+            superblock->SetStatus(SEEN_OBJECT_ERROR_INVALID);
+        }
+    }
+
+    // Remove triggers that are invalid or already executed
+    DBG( cout << "CGovernanceTriggerManager::CleanAndRemove: mapTrigger.size() = " << mapTrigger.size() << endl; );
+    trigger_m_it it = mapTrigger.begin();
+    while(it != mapTrigger.end())  {
+        bool remove = false;
+        CSuperblock_sptr& superblock = it->second;
+        if(!superblock)  {
+            DBG( cout << "CGovernanceTriggerManager::CleanAndRemove: NULL superblock marked for removal " << endl; );
+            remove = true;
+        }
+        else  {
+            DBG( cout << "CGovernanceTriggerManager::CleanAndRemove: superblock status = " << superblock->GetStatus() << endl; );
+            switch(superblock->GetStatus())  {
+            case SEEN_OBJECT_ERROR_INVALID:
+            case SEEN_OBJECT_UNKNOWN:
+                remove = true;
+                break;
+            case SEEN_OBJECT_EXECUTED:
+                {
```

Comment:
Because you need an extra block scope when declaring and assigning variables in a case.